src/gluonts/torch/model/tft/module.py

```diff
@@ -0,0 +1,371 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import Dict, List, Optional, Tuple
+
+import torch
+import torch.nn as nn
+from gluonts.core.component import validated
+from gluonts.torch.modules.quantile_output import QuantileOutput
+from gluonts.torch.modules.scaler import StdScaler
+from gluonts.torch.util import weighted_average
+
+from .layers import (
+    FeatureEmbedder,
+    FeatureProjector,
+    GatedResidualNetwork,
+    TemporalFusionDecoder,
+    TemporalFusionEncoder,
+    VariableSelectionNetwork,
+)
+
+
+class TemporalFusionTransformerModel(nn.Module):
+    """Temporal Fusion Transformer neural network.
+
+    Partially based on the implementation in github.com/kashif/pytorch-transformer-ts.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        context_length: int,
+        prediction_length: int,
+        d_past_feat_dynamic_real: List[int],
```

Comment:
I’m wondering whether it would make sense to have defaults for most of these. For example, in case one has no additional features (only target series) what would these be set to? I’m assuming some minimal values and then constant dummy tensors would be given to `forward` in place of features?

Reply:
Added defaults here + modified the docstring